From: Mauricio Faria de Oliveira <mauricfo@linux.vnet.ibm.com>
Date: Wed,  4 Jun 2014 09:06:27 -0300
Subject: ppc64el: kernel: patch: temporarily disable zImage

Debian ppc64el will wait for the 'powerpc/boot: 64bit little endian wrapper'
(zImage) patches upstream rather than shipping 32-bit tools for zImage.  The
patches are currently in Ben Herrenschmidt's linux-next tree [1].

The patches are not being carried over instead of this workaround because,
even without both, the build process does produces vmlinux, only failing for
zImage. So, it's ok to just work-around this for now, as we pick vmlinux.

The workaround patch just avoids this build error:

	[...]
	  LD      vmlinux
	  SYSMAP  System.map
	[...]
	  WRAP    arch/powerpc/boot/zImage.pseries
	ld: unrecognised emulation mode: elf32ppc
	Supported emulations: elf64lppc elf32lppc elf32lppclinux elf32lppcsim
	make[6]: *** [arch/powerpc/boot/zImage.pseries] Error 1
	make[5]: *** [zImage] Error 2

[1] https://git.kernel.org/cgit/linux/kernel/git/next/linux-next.git/log/?qt=grep&q=powerpc%2Fboot
    (15 commits with message prefix 'powerpc/boot:' dated of 2014-04-28)

Signed-off-by: Mauricio Faria de Oliveira <mauricfo@linux.vnet.ibm.com>
---
Index: linux-3.14.2/arch/powerpc/boot/Makefile
===================================================================
--- linux-3.14.2.orig/arch/powerpc/boot/Makefile	2014-05-02 16:52:37.000000000 -0300
+++ linux-3.14.2/arch/powerpc/boot/Makefile	2014-05-07 09:39:11.000000000 -0300
@@ -344,6 +344,15 @@
 $(obj)/%.dtb: $(src)/dts/%.dts FORCE
 	$(call if_changed_dep,dtc)
 
+# Debian ppc64el will wait for the 'powerpc/boot: 64bit little endian wrapper'
+# (zImage) patches upstream rather than shipping 32-bit tools for zImage.
+# This makes the next conditional block to pick vmlinux.strip as image-y.
+ifdef CONFIG_PPC64
+ifdef CONFIG_CPU_LITTLE_ENDIAN
+image-y :=
+endif
+endif
+
 # If there isn't a platform selected then just strip the vmlinux.
 ifeq (,$(image-y))
 image-y := vmlinux.strip
